lib/gate/coercer.rb

```diff
@@ -1,14 +1,10 @@
 module Gate
   class Coercer
     def initialize(engine, type, allow_nil: false)
-      unless Axiom::Types.const_defined?(type)
-        fail CoercionError, "Doesn't know how to coerce into #{type}"
-      end
-
       @engine = engine
       @type = type
       @allow_nil = allow_nil
-      @coercion_method = Axiom::Types.const_get(type).coercion_method
+      @coercion_method = set_coercion_method
```

Comment:
Actually you can remove this line and `:coercion_method` from `attr_readers` and just rename `set_coercion_method` to `coercion_method`.